### DISCUSSION TITLE
At what point / how is the onchain wallet created?

### DISCUSSION THREAD

**Discussion by we-do-it-lu** [Ride-The-Lightning/RTL#1278] [General]
I installed RTL. 
I have not created LN channels yet as first I'd like to understand when the onchain wallet / addressed it uses are generated: 
e.g. When I install BTCPayServer for instance I am asked to write down generated seed words which is used to derive adresses.
How are the addresses generated in RTL? Seed words? 
Thank you in advance.

**Comment by ShahanaFarooqui** [comment]
Which lightning implementation are you on?

**Reply by we-do-it-lu** [reply]
c-lightning

**Reply by we-do-it-lu** [reply]
https://www.reddit.com/r/TheLightningNetwork/comments/ugx0ep/clightning_installed_but_wheres_the_wallet/

No more questions txs.

**Reply by ShahanaFarooqui** [reply]
Great that you already found the answer. Here is another [link](https://github.com/ElementsProject/lightning/blob/98ad46795f2dfdfbef0d0df3b5d7a6ba7f142b1e/doc/beginners-guide/backup-and-recovery.md?plain=1#L440-L459) to explain hsm secret, wallet backup and wallet recovery
